PlantDiary001/Pages/Index.cshtml.cs

```diff
@@ -90,5 +84,15 @@ public void OnGet()
 
             
         }
+
+        public string GetData(string endpoint)
+        {
+            string downloadedData = "";
+            using (WebClient webClient = new WebClient())
+            {
+                downloadedData = webClient.DownloadString(endpoint);
+            }
+            return downloadedData;
```

Comment:
This was a really good addition, because you're reducing duplication!  I like it.

Reply:
Thanks!  This was really helpful.